services/rsources/handler.go

```diff
@@ -273,6 +273,50 @@ func (sh *sourcesHandler) Delete(ctx context.Context, jobRunId string, filter Jo
 	return tx.Commit()
 }
 
+func (sh *sourcesHandler) DeleteFailedRecords(ctx context.Context, jobRunId string, filter JobFilter) error {
```

Comment:
Should we migrate tests to the latest version as well? Anything that we can do so that deletion of v1 code is easier in the future

Reply:
True
I'll add some more with these new methods to this PR.
The older tests can be dropped in the future.